test/models/rss_log_test.rb

```diff
@@ -2,5 +2,44 @@
 
 require("test_helper")
 
+# tests of MO's "rss_logs"
 class RssLogTest < UnitTestCase
+  def test_url_for_normalized_controllers
+    normalized_rss_log_types.each do |type|
+      rss_log = rss_log_for_type(type)
+      id = rss_log.target_id
+      assert_match(type_normalized_show_path(type, id), rss_log.url,
+                   "rss_log.url incorrect for #{model(type)}")
+    end
+  end
+
+  # ---------- helpers ---------------------------------------------------------
+
+  def normalized_rss_log_types
+    RssLog.all_types.each_with_object([]) do |type, ary|
+      ary << type if model(type).controller_normalized?(model(type).name)
+    end
+  end
+
+  def model(type)
+    type.camelize.constantize
+  end
+
+  # rss_log factory
+  def rss_log_for_type(type)
+    # Target must have id; use an existing object to avoid hitting db
+    target = model(type).first
+    rss_log = RssLog.new
+    rss_log["#{type}_id".to_sym] = target.id
+    rss_log.updated_at = Time.zone.now
+    rss_log
+  end
+
+  def target_id(rss_log)
+    rss_log.target_id
+  end
+
+  def type_normalized_show_path(type, id)
```

Comment:
This seems a bit circular actually.  If I'm understanding this test correctly, the whole point is to test whether the model "knows" its own path correctly.  So you test that "model.show_path" = "model.show_controller/id"... but surely model.show_path uses model.show_controller internally(?)  (If not, then there is clearly room for DRY improvements.)  It seems to me this would be a more meaningful test if the test derived the correct path itself independently.  That is, convert type to the appropriate path.  Or perhaps better, just write in the expectation explicitly, then we're left with no question about whether it's making the correct, independent assertion.

Reply:
I don't think it's circular.
It compares
- the normalized show path for that type (`type_normalized_show_path(type, id)`)
to
- whatever is returned by RssLog.url (`rss_log.url`), which is hard coded and could be incorrect. 
For `article`, RssLog.url was returning `/article/show_article/...` which is wrong (and is fixed in this PR); it should be `/articles/...`